### ISSUE TITLE
Is there a way to extract the runtime statistics(node stats) shown in tensorboard?

### ISSUE THREAD

**Issue by allencho1222** [tensorflow/tensorboard#1028]
![image](https://user-images.githubusercontent.com/14326227/37155161-80111b80-2325-11e8-8c09-f3a7c09c06cf.png)

As we can see from this figure, after adding metadata, I can get the runtime statistics such as compute time and memory.
However, what I want to do is to extract these values(compute time, memory), and it seems that there's nothing on how we can extract it. Even it wasn't possible for me to how they are derived.

Is there any ways to extract these data from event file or other solutions?

**Comment by nfelt** [comment]
I don't think there's a way in TensorBoard itself to export this data, but it should be derived from a RunMetadata protocol buffer stored as an Event in event files in the log directory. 

https://github.com/tensorflow/tensorflow/blob/v1.6.0/tensorflow/core/util/event.proto#L36
https://github.com/tensorflow/tensorflow/blob/r1.6/tensorflow/core/protobuf/config.proto#L401

It should be possible to write a python script that processes the event files and detects instances of TaggedRunMetadata which you could then deserialize into a RunMetadata and output node-specific runtime statistics.  See e.g. 
https://www.tensorflow.org/api_docs/python/tf/train/summary_iterator

**Comment by chihuahua** [comment]
Indeed, like @nfelt noted, you can write a script to parse `RunMetadata` from events files.

Another thing you can do just from the TensorBoard application is 

1. Open the Network panel within [Chrome Developer Tools](https://developers.google.com/web/tools/chrome-devtools/).
2. Visit TensorBoard's graphs dashboard. Select a session run (assuming your model logic collected metadata for at least 1 session run).
3. Note the response to the `//run_metadata` endpoint within the Network panel. The response is text content that represents the RunMetadata in "pbtxt" format for a specific run and tag.
4. Code some logic that parses that text response.

**Comment by bileschi** [comment]
Even though there is a work-around, leaving this open as a place where the command-line API for accessing serialized data
